lms/static/scripts/frontend_apps/utils/api.js

```diff
@@ -90,6 +94,36 @@ async function apiCall({ path, authToken, data }) {
   return resultJson;
 }
 
-// Separate export from declaration to work around
-// https://github.com/robertknight/babel-plugin-mockable-imports/issues/9
-export { apiCall };
+/** @param {number} ms */
+function delay(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+/**
+ * Fetch a list of available ebooks to use in assignments.
+ *
+ * @param {string} authToken
+ * @param {number} [fetchDelay] - Dummy delay to simulate slow third-party
+ * @return {Promise<Book[]>}
+ */
+export async function fetchBooks(authToken, fetchDelay = 500) {
```

Comment:
This is totally the kitchen drawer folder `utils`, and may need some TLC later. 

I realize these are placeholder, but when we replace them with the real calls, it may make sense to move them to their own file like the google api calls, grader-service, etc.

Reply:
> This is totally the kitchen drawer folder utils, and may need some TLC later.

Yes. I think it may be helpful to create a service layer in the LMS frontend and create a stateful service for API calls. That will enable us to avoid passing around the `authToken` to every function that makes API calls.

Reply:
I wonder if it might be worth extracting these in-flux VitalSource-related API mocks (as it were) into a separate module, even now? Their presence here in their current prototype states feels at odds with the production-class code in this module. I don't feel strongly about this, however.

Reply:
I wrote about this in a separate comment, but I think there will be some refactoring of other API-calling code that we might want to do at the same time for consistency. For example, we might require all calls to the API to go through wrapper functions. I've left this as a follow-up exercise for the moment.